src/Symfony/Component/Security/Http/AccessToken/Oidc/OidcTokenHandler.php

```diff
@@ -111,4 +97,92 @@ public function getUserBadgeFrom(string $accessToken): UserBadge
+            new Checker\AlgorithmChecker($this->signatureAlgorithm->list()),
+        ], [
+            new JWSTokenSupport(),
+        ]);
+        // if this check fails, an InvalidHeaderException is thrown
+        $headerCheckerManager->check($jws, 0);
+
+        return json_decode($jws->getPayload(), true);
+    }
+
+    private function verifyClaims(array $claims): array
+    {
+        // Verify the claims
+        $checkers = [
+            new Checker\IssuedAtChecker(clock: $this->clock, allowedTimeDrift: 0, protectedHeaderOnly: true),
+            new Checker\NotBeforeChecker(clock: $this->clock, allowedTimeDrift: 0, protectedHeaderOnly: true),
+            new Checker\ExpirationTimeChecker(clock: $this->clock, allowedTimeDrift: 0, protectedHeaderOnly: true),
+            new Checker\AudienceChecker($this->audience),
+            new Checker\IssuerChecker($this->issuers),
+        ];
+        $claimCheckerManager = new ClaimCheckerManager($checkers);
+
+        // if this check fails, an InvalidClaimException is thrown
+        return $claimCheckerManager->check($claims);
+    }
+
+    private function decryptIfNeeded(string $accessToken): string
+    {
+        if (null === $this->decryptionKeyset || null === $this->decryptionAlgorithms) {
+            $this->logger?->debug('The encrypted tokens (JWE) are not supported. Skipping.');
+
+            return $accessToken;
+        }
+
+        $jweHeaderChecker = new Checker\HeaderCheckerManager(
+            [
+                new Checker\AlgorithmChecker($this->decryptionAlgorithms->list()),
+                new Checker\CallableChecker('enc', fn ($value) => \in_array($value, $this->decryptionAlgorithms->list())),
+                new Checker\CallableChecker('cty', fn ($value) => 'JWT' === $value),
+                new Checker\IssuedAtChecker(clock: $this->clock, allowedTimeDrift: 0, protectedHeaderOnly: true),
+                new Checker\NotBeforeChecker(clock: $this->clock, allowedTimeDrift: 0, protectedHeaderOnly: true),
+                new Checker\ExpirationTimeChecker(clock: $this->clock, allowedTimeDrift: 0, protectedHeaderOnly: true),
+            ],
+            [new JWETokenSupport()]
+        );
+        $jweDecrypter = new JWEDecrypter($this->decryptionAlgorithms, null);
+        $serializerManager = new JWESerializerManager([new JweCompactSerializer()]);
+        try {
+            $jwe = $serializerManager->unserialize($accessToken);
+            $jweHeaderChecker->check($jwe, 0);
+            $result = $jweDecrypter->decryptUsingKeySet($jwe, $this->decryptionKeyset, 0);
+            if (false === $result) {
+                throw new \Exception('The JWE could not be decrypted.');
+            }
+
+            $payload = $jwe->getPayload();
+            if (null === $payload) {
+                throw new \Exception('The JWE payload is empty.');
+            }
+
+            return $payload;
+        } catch (\Throwable $e) {
```

Comment:
Could it catch more precise exception types? Catch-all should generally be avoided